src/sagemaker_training/files.py

```diff
@@ -48,7 +50,11 @@ def write_failure_file(failure_msg):  # type: (str) -> None
         failure_msg: The description of failure.
     """
     file_path = os.path.join(environment.output_dir, "failure")
-    write_file(file_path, failure_msg)
+
+    # Write failure file only if it does not exist
+    if not os.path.exists(file_path):
+        logger.info("Failure file already exists. Skipping creation....")
```

Comment:
Shouldn't this be logged when the above returns true?

Reply:
Updated.